leakcanary-android-core/src/main/java/leakcanary/NotificationEventListener.kt

```diff
@@ -5,24 +5,44 @@ import android.app.NotificationManager
 import android.app.PendingIntent
 import android.content.Context
 import android.os.Build
+import android.os.Handler
+import android.os.HandlerThread
 import com.squareup.leakcanary.core.R
 import leakcanary.EventListener.Event
 import leakcanary.EventListener.Event.DumpingHeap
 import leakcanary.EventListener.Event.HeapAnalysisDone
 import leakcanary.EventListener.Event.HeapAnalysisDone.HeapAnalysisSucceeded
 import leakcanary.EventListener.Event.HeapAnalysisProgress
-import leakcanary.EventListener.Event.HeapDumpFailed
 import leakcanary.EventListener.Event.HeapDump
+import leakcanary.EventListener.Event.HeapDumpFailed
 import leakcanary.internal.InternalLeakCanary
+import leakcanary.internal.NotificationReceiver
 import leakcanary.internal.NotificationType.LEAKCANARY_LOW
 import leakcanary.internal.NotificationType.LEAKCANARY_MAX
 import leakcanary.internal.Notifications
+import leakcanary.internal.RetainInstanceEvent
+
+private const val BACKGROUND_THREAD_NAME = "Leakcanary-Notification-Event-listener"
+private const val DISMISS_NO_RETAINED_OBJECT_NOTIFICATION_MILLIS = 30_000L
 
 object NotificationEventListener : EventListener {
 
   private val appContext = InternalLeakCanary.application
   private val notificationManager =
     appContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+  private val backgroundHandler: Handler
+  private val scheduleDismissNoRetainedOnTapNotification = {
+    dismissNoRetainedOnTapNotification()
+  }
+  private val scheduleDismissRetainedCountNotification = {
+    dismissRetainedCountNotification()
+  }
+
+  init {
+    val handlerThread = HandlerThread(BACKGROUND_THREAD_NAME)
+    handlerThread.start()
+    backgroundHandler = Handler(handlerThread.looper)
```

Comment:
I don't really get why we'd need a new thread rather than just main thread. Previous code was probs piggy backing on handy handler.

Reply:
I thought we were using the background thread for performance reasons.